carmm/run/workflows/react.py

```diff
@@ -579,32 +580,36 @@ def _calc_generator(params,
     from carmm.run.aims_calculator import get_aims_and_sockets_calculator
     """On machines where ASE and FHI-aims are run separately (e.g. ASE on login node, FHI-aims on compute nodes)
     we need to specifically state what the name of the login node is so the two packages can communicate"""
-    sockets_calc, fhi_calc = get_aims_and_sockets_calculator(dimensions=dimensions,
-                                                             logfile=f"{directory}/socketio.log",
-                                                             verbose=True,
-                                                             codata_warning=False,
-                                                             directory=directory
-                                                             )
 
     """Remove previous xc argument to ensure libxc warning override is first"""
-    fhi_calc.parameters = {k: v for k, v in fhi_calc.parameters.items() if k != 'xc'}
-    fhi_calc.parameters['override_warning_libxc'] ='True'
+    parameters = dict(params)
+    xc = parameters['xc']
+    del parameters['xc']
+    if "libxc" in xc:
+        parameters['override_warning_libxc'] ='True'
+    parameters['xc'] = xc
```

Comment:
Maybe I'm missing something, but these seem to do the same thing and the only change is that you work with a local variable, instead of the FHI-aims calculator object. 

The prior code is much more succinct, and so it makes more sense to me for this to persist. Can you explain what the benefit is from this change?

Reply:
By dealing with a local variable `parameters`, we can avoid messing with fhi_calc.parameters which should provide compatibility for both ase 3.22 and 3.23.

Reply:
Can we use an instance of the class then, rather than the workaround? Presumably one can make a fresh parameters object and pass the `**dictionary ` to it as keyword arguments.

Reply:
Would something like:
`fhi_calc.parameters.__dict__ = {k: v for k, v in fhi_calc.parameters.__dict__ if k != 'xc'}`
work? We do not alter the class then, and the solution should be Class-agnostic

Reply:
if 3.23.0 is a dictionary and not a class, then it will not work I guess

Reply:
we could have a `type(fhi_calc.parameters)`  or ASE version check in place and use either

`fhi_calc.parameters.__dict__ = {k: v for k, v in fhi_calc.parameters.__dict__ if k != 'xc'}`
or
`fhi_calc.parameters = {k: v for k, v in fhi_calc.parameters.items() if k != 'xc'}`

Reply:
I say keep all keyword alterations as they are, add an ASE 3.22.1 conditional import `from ase.calculators.calculator import Parameters` and before the calculator is returned, have an ASE 3.22.1 conditional statement to put the dictionary into `fhi_calc.parameters = Parameters(**fhi_calc.parameters)`